docs(blog): update ts post (#6027)
refinedev/refine: documentation/blog/2024-06-10-next-typescript.md

slug: next-js-with-typescript
authors: michael
tags: [nextjs, typescript]
image: https://refine.ams3.cdn.digitaloceanspaces.com/blog/2022-09-12-next-typescript/social-3.png
hide_table_of_contents: false
---

> **_This article was last updated on Jun 10, 2024 to add new section for ts.config and add some tips on Next.js Typescript_**

Reviewer: ```suggestion
**_This article was last updated on Jun 10, 2024 to add new section for tsconfig.json and add some tips on usage with Next.js_**
```

export default Profile;
```

### Understanding tsconfig.json in Next.js

> The `tsconfig.json` is the main piece of configuration for TypeScript with Next.js. This file contains all the compiler options and settings for the TypeScript compiler to know exactly how you want your code transpiled. Here's a structure, which the typical `tsconfig.json` should have:

Reviewer: ```suggestion
The `tsconfig.json` is the main piece of configuration for TypeScript. This file contains all the compiler options and settings for the TypeScript compiler to know exactly how you want your code transpiled. Here's a structure, which the typical `tsconfig.json` should have:
```

   ```
   /src
     /parts
     /pages
     /helpers
     /styles
     /services
   ```

2. **Use a Common Naming Convention:**
   Follow the file and folder naming convention. One such example is the naming of files in camelCase and React component names in PascalCase.

   ```
   /components
     header.ts
     Profile.ts
   ```

3. **Make it Human:**
   Modularize your code by separation of concerns. Put your business logic in service files and keep your components' functionality focused on the presentation.
   ```
   /services
     userService.ts
   /parts
     UserProfile.tsx
   ```

#### Writing Clean, Maintainable Code

Clean and maintainable code ensures better readability and less technical debt. Here are some of the best practices for writing clean TypeScript code in Next.js:

### Type Annotations

Types should be declared expressly when declaring function parameters, return values, and variables. This makes your code predictable and more accessible to follow.

```typescript
const add = (a: number, b: number): number => {
  return a + b;
};
```

### Do Not Use `any`

> Do not use `any` at all. It is the same as using TypeScript for nothing—it is better to use some types or generics.

Reviewer: ```suggestion
Do not use `any` at all. It is the same as not using TypeScript. Better to use some types or generics.
```
